src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,712 @@
 #include <boost/wave/cpplexer/cpp_lex_token.hpp>
 #include <boost/wave/cpplexer/cpp_lex_iterator.hpp>
 
-#define TCPP_IMPLEMENTATION
-#include <tcpp/source/tcppLibrary.hpp>
-#undef TCPP_IMPLEMENTATION
-
 using namespace nbl;
 using namespace nbl::asset;
 using Microsoft::WRL::ComPtr;
 
 static constexpr const wchar_t* SHADER_MODEL_PROFILE = L"XX_6_7";
 
-namespace nbl::asset::hlsl::impl
+
+namespace nbl::wave {
+    using namespace boost;
+    using namespace boost::wave;
+    using namespace boost::wave::util;
+
+
+    class include_paths
+    {
+    private:
+        typedef std::list<std::pair<system::path, std::string> >
+            include_list_type;
+        typedef include_list_type::value_type include_value_type;
+
+#if BOOST_WAVE_SUPPORT_PRAGMA_ONCE != 0
+        typedef boost::wave::util::bidirectional_map<std::string, std::string>::type
+            pragma_once_set_type;
+#endif
+
+    public:
+        include_paths()
+            : was_sys_include_path(false),
+            current_dir(),
+            current_rel_dir()
+        {}
+
+        bool add_include_path(char const* path_, bool is_system = false)
+        {
+            return add_include_path(path_, (is_system || was_sys_include_path) ?
+                system_include_paths : user_include_paths);
+        }
+
+        void set_sys_include_delimiter() { was_sys_include_path = true; }
+
+        bool find_include_file(std::string& s, std::string& dir, bool is_system,
+            char const* current_file) const;
+        system::path get_current_directory() const
+        {
+            return current_dir;
+        }
+        void set_current_directory(char const* path_) {
```

Comment:
DOCUMENTATION!

Write that:
- `path_` is supposed to be a path-to-file not directory!
- `path_` is supposed to be relative and not absolute?